sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2404,6 +2409,19 @@ pi_result cuda_piKernelCreate(pi_program program, const char *kernel_name,
   }
 
   *kernel = retKernel.release();
+
+  {
+    pi_device device = program->get_context()->get_device();
+    size_t reqdThreadsPerBlock[3] = {};
+    pi_result retError = cuda_piKernelGetGroupInfo(
+        *kernel, device, PI_KERNEL_GROUP_INFO_COMPILE_WORK_GROUP_SIZE,
+        sizeof(reqdThreadsPerBlock), reqdThreadsPerBlock, nullptr);
+    assert(retError == PI_SUCCESS);
+
+    (*kernel)->save_reqdThreadsPerBlock(sizeof(reqdThreadsPerBlock),
+                                        reqdThreadsPerBlock);
+  }
```

Comment:
What is the motivation for giving this its own scope?

Reply:
there was no motivation other than to separate this code, I removed it to avoid confusion.